src/crypto/DeviceList.js

```diff
@@ -542,15 +705,13 @@ class DeviceListUpdateSerialiser {
             this._olmDevice, userId, userStore, response || {},
         );
 
-        // update the session store
+        // put the updates into thr object that will be returned as our results
         const storage = {};
         Object.keys(userStore).forEach((deviceId) => {
             storage[deviceId] = userStore[deviceId].toStorage();
         });
 
-        this._sessionStore.storeEndToEndDevicesForUser(
-            userId, storage,
-        );
+        this._deviceList._setRawStoredDevicesForUser(userId, storage);
```

Comment:
I'm not sure this is any more elegant than just manipulating the device list directly, but fine

Reply:
Probably not, but it seemed prudent to at least do it via setters & getters.